src/openrct2-ui/windows/Park.cpp

```diff
@@ -1495,6 +1511,199 @@ static void window_park_objective_paint(rct_window* w, rct_drawpixelinfo* dpi)
         dpi, ft.Data(), screenCoords, 221, ObjectiveNames[gScenarioObjective.Type], COLOUR_BLACK);
     screenCoords.y += 5;
 
+    // Your progress
+    // if (gScenarioObjective.Check() == ObjectiveStatus::Undecided)
+    if (gScenarioCompletedCompanyValue == MONEY32_UNDEFINED)
+    {
+        gfx_draw_string_left(dpi, STR_OBJECTIVE_PROGRESS_LABEL, nullptr, COLOUR_BLACK, screenCoords);
+        screenCoords.y += LIST_ROW_HEIGHT;
+
+        // Progress
+        int32_t barColour = COLOUR_BRIGHT_GREEN;
+
+        switch (gScenarioObjective.Type)
+        {
+            case OBJECTIVE_NONE:
+            case OBJECTIVE_HAVE_FUN:
+            case OBJECTIVE_BUILD_THE_BEST:
+                screenCoords.y += gfx_draw_string_left_wrapped(
+                    dpi, nullptr, screenCoords, 221, STR_OBJECTIVE_PROGRESS_NONE, COLOUR_BLACK);
+                break;
+            case OBJECTIVE_GUESTS_BY:
+            case OBJECTIVE_GUESTS_AND_RATING:
+            {
+                float guestsInParkPercentage
+                    = (static_cast<float>(gNumGuestsInPark) / static_cast<float>(gScenarioObjective.NumGuests));
+                float parkRatingPercentage;
+
+                ft = Formatter();
+                ft.Add<uint32_t>(gNumGuestsInPark);
+                ft.Add<uint32_t>(static_cast<uint32_t>(gScenarioObjective.NumGuests));
+
+                screenCoords.y += gfx_draw_string_left_wrapped(
+                    dpi, ft.Data(), screenCoords, 221, STR_OBJECTIVE_PROGRESS_GUESTS, COLOUR_BLACK);
+                screenCoords.y += 1;
+
+                window_park_objective_bars_paint(guestsInParkPercentage, screenCoords.x, screenCoords.y, w, dpi, barColour);
+                screenCoords.y += LIST_ROW_HEIGHT;
+
+                ft = Formatter();
+                ft.Add<uint32_t>(static_cast<uint32_t>(gParkRating));
+
+                if (gScenarioObjective.Type == OBJECTIVE_GUESTS_BY)
+                {
+                    ft.Add<uint32_t>(static_cast<uint32_t>(600));
+                    parkRatingPercentage = (static_cast<float>(gParkRating) / static_cast<float>(600));
```

Comment:
What's the `600` magic number? Maybe extract it to a `constexpr` for better documentation

Reply:
The 600 and 700 magic number comes from Scenario.cpp. It's used in the functions CheckGuestsBy and CheckGuestsAndRating. I agree a constant would be better. I thought that the whole code of Scenario.cpp could be overhauled, but I guess a step between won't hurt anyone.

Reply:
Replaced 600 by the constexpr OBJECTIVE_GUESTS_BY_PARK_RATING.

Reply:
Nice!